pkg/providers/v1/aws_ec2.go

```diff
@@ -30,6 +30,16 @@ type awsSdkEC2 struct {
 	ec2 ec2iface.EC2API
 }
 
+func (s *awsSdkEC2) DescribeInstanceTopology(request *ec2.DescribeInstanceTopologyInput) (*ec2.InstanceTopology, error) {
+	resp, err := s.ec2.DescribeInstanceTopology(request)
+	if err != nil {
+		return nil, fmt.Errorf("error describe AWS Instance Topology: %q", err)
+	} else if len(resp.Instances) == 0 {
+		return nil, nil
+	}
+	return resp.Instances[0], err
```

Comment:
this wrapper's logic seems strange, shouldn't we return the list of `[]*ec2.InstanceTopology` instead of just the first item?

Reply:
DescribeInstanceTopology can accept a list of instance IDs, if we pass only 1 instance ID, it will return 1 according topology info something like below.
```
"Instances": [
        {
            "InstanceId": "i-xxx",
            "InstanceType": "p5e.48xlarge",
            "NetworkNodes": [
                "nn-xxxxx",
                "nn-xxxxx",
                "nn-xxxxx"
            ],
            "AvailabilityZone": "us-west-2c",
            "ZoneId": "usw2-az3"
        }
    ]
```
But yeah, it seems more reasonable to return all instances in api and get the first instance in where trigger it and get result. Will revised in next

Reply:
agreed, because we still allow the full `ec2.DescribeInstanceTopologyInput` payload which could specify more than one instance id

Reply:
revised